app/api.py

```diff
@@ -0,0 +1,195 @@
+import uvicorn
+from fastapi import FastAPI, File, UploadFile, HTTPException, Form, Request, Depends
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel
+from typing import List, Dict, Any, Optional
+from contextlib import asynccontextmanager
+import shutil
+from fuzzywuzzy import fuzz, process
+from doctr.models import db_resnet50, crnn_vgg16_bn
+from app.doctr_module import load_custom_model, extract_names_and_boxes
+from app.pdf_ocr import extract_text_and_boxes_from_pdf
+from app.vector_database import store_text_in_vector_db
+from app.rag import retrieve_answer
+from config.config import logger
+from pathlib import Path
+import shutil
+
+
+app = FastAPI()
+
+# Define paths for the model weights
+det_weights_path = "models/db_resnet50-79bd7d70.pt"
+reco_weights_path = "models/crnn_vgg16_bn-9762b0b0.pt"
+
+# Global variables to hold the models
+det_model = None
+reco_model = None
+
+
+# Define a model for input data
+class NamePairs(BaseModel):
+    file_location: Optional[str] = None
+    names: Optional[list] = []
+
+
+# Custom dependency to handle form data and JSON
+async def get_name_pairs(
+    request: Request,
+    file_location: Optional[str] = Form(None),
+    names: Optional[str] = Form(None),
+    file: Optional[UploadFile] = File(None),
+) -> NamePairs:
+    content_type = request.headers.get("content-type")
```

Comment:
I am not sure if it is a good idea to set up an endpoint to receive both form and JSON data, it will mess up the single responsibility principle. I will suggest creating two different endpoints for the two use cases. The protocols to parse multiform data are different than the JSON and the Documentation from FAstAPI does no encourage mixing those in a single endpoint.